stablehlo/transforms/ChloDecompositionPatterns.td

```diff
@@ -46,6 +46,39 @@ def StableHLO_ConstantLikeSmallestNormalizedValue : NativeCodeCall<
 // Unary op patterns.
 //===----------------------------------------------------------------------===//
 
+// Expand acos for non-complex arguments to MHLO dialect as follows:
+//   acos(x) = 2 * atan2(sqrt(1 - x^2), (1 + x))  if x != -1
```

Comment:
Using `1-x^2` will be inaccurate when `abs(x)` is close to `1`. The corresponding cancellation errors are avoided when using `(1-x)*(1+x)`. I can address this in a follow-up PR if needed.

Reply:
Created https://github.com/openxla/stablehlo/issues/2452 to track this work, in the meantime this can be seen as a revert